application-admintools-default/src/main/java/com/xwiki/admintools/internal/AdminToolsManager.java

```diff
@@ -78,14 +91,14 @@ public String generateData()
      * @param hint {@link String} represents the data provider identifier.
      * @return a {@link String} representing a template
      */
-    public String generateData(String hint)
+    public String generateData(String hint) throws ComponentLookupException
     {
-        for (DataProvider dataProvider : this.dataProviderProvider.get()) {
-            if (dataProvider.getIdentifier().equals(hint)) {
-                return dataProvider.getRenderedData();
-            }
+        DataProvider dataProvider = contextComponentManager.getInstance(DataProvider.class, hint);
+        if (dataProvider == null) {
```

Comment:
The component manager should never return null. If it can't find the requested component then it throws ``ComponentLookupException``.